app/assets/javascripts/swagger-ui/extensions.coffee

```diff
@@ -39,12 +39,10 @@ ThreeScaleAutoComplete.handleData = (data)->
   ThreeScaleAutoComplete.DataStatus = data.status
   if(+data.status == 200)
     template = Handlebars.compile(ThreeScaleAutoComplete.TipTemplate)
-    html = _.reduce(data.results
-      , (memo, values, key)->
-          values = {type: key, items: values, description: ThreeScaleAutoComplete.TipDescriptions[key]}
-          memo + template(values)
-      , "")
-    $("body").append(html)
+    html = for key, value of data.results
+      template { type: key, items: value, description: ThreeScaleAutoComplete.TipDescriptions[key] }
```

Comment:
hm, seems that the original code provides a "sum" of the `template` results, not sure if this is the case with the new code? :thinking:

Reply:
Right, this involves a bit of coffeescript magic. It's called [comprehensions](https://coffeescript.org/#loops) and it's basically a "clever" way to do loops.

In this case, the loop will map `data.results` into an array of `template(...)` (an array of strings). Then the strings is turned into 1 with `.join` in line 45.

Original code would start with an empty string and iterate over `data.results`, adding each time a new segment to the final string: `memo + template()`. Each following iteration, memo's value is the one returned by the previous iteration (standard reduce behavior).

Does it make more sense now?

Reply:
Thank you for the clarification! Yeah, I was just kind of lost in this weird loop... 
Maybe at some point we can get rid of :coffee: too :grimacing: